src/store/modules/user/actions.ts

```diff
@@ -130,13 +130,20 @@ const actions: ActionTree<UserState, RootState> = {
   /**
    * update current eComStore information
    */
-  async setEcomStore({ commit, dispatch }, payload) {
+  async setEcomStore({ commit, dispatch, getters }, payload) {
     dispatch('job/clearJobState', null, { root: true });
-    commit(types.USER_CURRENT_ECOM_STORE_UPDATED, payload.eComStore);
-    dispatch('getShopifyConfig', payload.eComStore.productStoreId);
+    let productStore;
+
+    if(payload.productStore) {
+      commit(types.USER_CURRENT_ECOM_STORE_UPDATED, payload.productStore);
```

Comment:
We should commit in both the cases, here we should assign productStore with payload.productStore